packages/mui-material/src/Select/SelectInput.js

```diff
@@ -418,10 +418,10 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
 
   if (computeDisplay) {
     if (multiple) {
-      if (value.length === 0) {
-        display = '';
+      if (displayMultiple.length === 0) {
```

Comment:
Using value here could create more bugs. `value` !== `displayMultiple`.

Reply:
I agree. Proof that the error no longer exists with this build: https://codesandbox.io/s/multipleselect-material-demo-forked-nefti?file=/package.json